vscode-antimony/src/server/bioservices_server/webservices.py

```diff
@@ -92,3 +114,55 @@ def annot_search_uniprot(self, query: str):
                 'prefix': 'uniprot',
             })
         return objects
+    
+    def annot_search_rhea(self, query: str):
+        self.init_rhea()
+        
+        if query.strip() == '':
+            return list()
+
+        try:
+            result_df = self.rhea.search(query, columns='rhea-id,equation')
+        except URLError:
+            raise NetworkError
+
+        result_df = result_df[0:20]
+        result_l = result_df.values.tolist()
+        objects = list()
+        for row in result_l:
+            id_ = row[0]
+            equation = row[1]
+            id_ = id_[5:]
+            objects.append({
+                'id': id_,
+                'name': equation,
+                'detail': id_,
+                'prefix': 'rhea',
+            })
+        return objects
+    
+    def annot_search_ontology(self, query: str, ontology_id: str):
+        self.init_ontology()
+        ontology_id = ontology_id.lower()
+        
+        if query.strip() == '':
+            return list()
+        try:
+            result_dict = self.ontology.search(query, ontology_id)
+            result_dicts = result_dict['response']['docs']
+        except URLError:
+            raise NetworkError
+        
+        objects = list()
+        
+        for d in result_dicts:
+            iri = d['iri']

```

Comment:
It's best practice to define global variables for literal constants. For example, define IRI = 'iri" at the top of the module and then use "iri = d[IRI]".

Also, although it's good to have short names for loop variables, a one character name is an easy source of bugs (e.g.., with other 1 character names) and is less readable. consider "for dct in result_dicts" instead.